extensions/trace_propagators/src/main/java/io/opentelemetry/extensions/trace/propagation/JaegerPropagator.java

```diff
@@ -187,6 +208,42 @@ public static JaegerPropagator getInstance() {
     return buildSpanContext(traceId, spanId, flags);
   }
 
+  private static <C> Baggage getBaggageFromHeader(C carrier, Getter<C> getter) {
+    Baggage.Builder builder = null;
+
+    for (String key : getter.keys(carrier)) {
+      if (key.startsWith(BAGGAGE_PREFIX)) {
+        if (key.length() <= BAGGAGE_PREFIX.length()) {
+          continue;
+        }
+
+        if (builder == null) {
+          builder = Baggage.builder();
+        }
+        builder.put(key.substring(BAGGAGE_PREFIX.length()), getter.get(carrier, key));
+      } else if (key.equals(BAGGAGE_HEADER)) {
+        builder = parseBaggageHeader(getter.get(carrier, key), builder);
+      }
+    }
+    return builder == null ? null : builder.build();
+  }
+
+  @SuppressWarnings("StringSplitter")
+  private static Baggage.Builder parseBaggageHeader(String header, Baggage.Builder builder) {
+    for (String part : header.split("\\s*,\\s*")) {
```

Comment:
Regex split looks like it could be very slow. Best could be to use indexOf to avoid split array allocation but if too tedious can we at least just split on the single chars?

Reply:
Let's do that in a follow-up. This code is ported directly from the Jaeger Propagator, but yes, we could try to optimize things in follow ups (we did the same for OC's TraceContext propagator, for example).